metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -172,20 +172,30 @@ ControllerResult<Map<ConfigResource, ApiError>> incrementalAlterConfigs(
         Map<ConfigResource, ApiError> outputResults = new HashMap<>();
         for (Entry<ConfigResource, Map<String, Entry<OpType, String>>> resourceEntry :
                 configChanges.entrySet()) {
-            incrementalAlterConfigResource(resourceEntry.getKey(),
+            ApiError apiError = incrementalAlterConfigResource(resourceEntry.getKey(),
                 resourceEntry.getValue(),
                 newlyCreatedResource,
-                outputRecords,
-                outputResults);
+                outputRecords);
+            outputResults.put(resourceEntry.getKey(), apiError);
         }
         return ControllerResult.atomicOf(outputRecords, outputResults);
     }
 
-    private void incrementalAlterConfigResource(ConfigResource configResource,
+    ControllerResult<ApiError> incrementalAlterConfig(
+        ConfigResource configResource, Map<String, Entry<OpType, String>> keyToOps,
```

Comment:
nit: move second argument to next line. Not everyone agrees, but I also like to put the closing parenthesis on a new line.
```java
ControllerResult<ApiError> incrementalAlterConfig(
    ConfigResource configResource, 
    Map<String, Entry<OpType, String>> keyToOps,
    boolean newlyCreatedResource
) {
...
```

Reply:
Changed to that. I agree the closing parenthesis on the next line reads a bit better.